integration-tests/api/__tests__/admin/customer-groups.js

```diff
@@ -86,6 +86,84 @@ describe("/admin/customer-groups", () => {
     })
   })
 
+  describe("POST /admin/customer-groups/:id", () => {
+    beforeEach(async () => {
+      try {
+        await adminSeeder(dbConnection)
+        await customerSeeder(dbConnection)
+      } catch (err) {
+        console.log(err)
+        throw err
+      }
+    })
+
+    afterEach(async () => {
+      const db = useDb()
+      await db.teardown()
+    })
+
+    it("updates group name & metadata", async () => {
+      const api = useApi()
+
+      const id = "customer-group-2"
+
+      const body = {
+        name: "vip-customers-v2",
+        metadata: {
+          metaKey1: `metaValue1`,
+        },
+      }
+
+      const response = await api.post(`/admin/customer-groups/${id}`, body, {
+        headers: {
+          Authorization: "Bearer test_token",
+        },
+      })
+
+      expect(response.status).toEqual(200)
+      expect(response.data.customerGroup).toEqual(
+        expect.objectContaining({
+          id: "customer-group-2",
+          name: "vip-customers-v2",
+          metadata: {
+            metaKey1: `metaValue1`,
+          },
+        })
+      )
+      expect(response.data.customerGroup).not.toHaveProperty("customers")
+    })
+
+    it("reset `metadata`to `null`", async () => {
+      const api = useApi()
+
+      const id = "customer-group-2"
+      // already has some metadata initially
+
+      const body = {
+        name: "vip-customers-v2",
+        metadata: null,
+      }
+
+      const response = await api
+        .post(`/admin/customer-groups/${id}?expand=customers`, body, {
+          headers: {
+            Authorization: "Bearer test_token",
+          },
+        })
+        .catch(console.log)
+
+      expect(response.status).toEqual(200)
+      expect(response.data.customerGroup).toEqual(
+        expect.objectContaining({
+          id: "customer-group-2",
+          name: "vip-customers-v2",
+          metadata: null,
+          customers: [],
+        })
+      )
+    })
+  })
```

Comment:
I don't think that this should be possible actually (see my comment below). If someone wants to "reset" metadata they should actively be setting each existing field to `null`. I.e. if the metadata is `{ keya: "something", keyb: "something" }` the update should be:

```
{
  metadata: { keya: null, keyb: null }
}
```